Implemented Multi-banners in HomeFragment

madim opened the pull request:
Implemented multi-banners in HomeFragment
Moved asking for location and bluetooth to the SettingsFragment
Cleaned up HomeFragment and HomeViewModel
Fixed scrolling issue in MenuFragment
Check if user tested positive every time we open HomeFragment

![image](https://user-images.githubusercontent.com/13729861/80272749-f7bdbe80-86ed-11ea-9743-5c636d1bea77.png)

madim committed:
Check if user tested positive every time we open HomeFragment

Also fixed scrolling issue in MenuFragment

nkumarcc commented on app/src/main/java/org/covidwatch/android/presentation/home/HomeViewModel.kt:
```diff
@@ -48,8 +48,6 @@ class HomeViewModel(
     private val interactedWithInfectedObserver = Observer<Boolean> { hasPossiblyInteractedWithInfected ->
```
Isn't it nicer for these bits of logic to live next to each other? It should be one banner or the other

madim replied:
If current user is tested positive should depend on `Observer` that listens to `contact_event` changes in db

This `if (hasPossiblyInteractedWithInfected && !isUserTestedPositive) {` makes sure those banners will not appear at the same time

nkumarcc commented on app/src/main/res/layout/fragment_menu.xml:
```diff
@@ -1,43 +1,50 @@
-        android:text="@string/menu_footer_text"
-        android:textAlignment="center"
-        android:textSize="12sp" />
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/menu_list"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="16dp"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager" />
```
Is this in case we have 2 banners?

madim replied:
No. It's just to fix to scrolling issue in Menu Screen for smaller screens. It's not related to the linked issue :)

nkumarcc reviewed (changes requested):
I have written thoughts

madim committed:
Bluetooth banner is more important than "Tested positive" banner

madim committed:
Implemented multi-banners in HomeFragment

Moved asking for location and bluetooth to the SettingsFragment
Cleaned up HomeFragment and HomeViewModel

jamesjmtaylor reviewed (approved):
LGTM.  On an unrelated note, has anyone noticed that it takes a full minute before the app begins registering touch events?  This is occurring on `develop` as well as this branch.  I found this in my logcat logs:
 ```
2020-04-25 08:32:29.628 32672-32672/org.covidwatch.tags.android I/Choreographer: Skipped 5378 frames!  The application may be doing too much work on its main thread.
```

redbassett commented on app/src/main/java/org/covidwatch/android/presentation/home/Banner.kt:
```diff
@@ -1,22 +1,17 @@
-    object TurnOnBluetooth : BannerAction()
-    object PotentialRisk : BannerAction()
-    object NoAction : BannerAction()
+sealed class InfoBanner {
```
This is a minor semantics nitpick, but for clarity, could this be titled `InfoBannerState` or something similar? Without looking directly at the subclasses this name doesn't tell me what this set of classes does.

madim commented:
> LGTM. On an unrelated note, has anyone noticed that it takes a full minute before the app begins registering touch events? This is occurring on `develop` as well as this branch. I found this in my logcat logs:
> 
> ```
> 2020-04-25 08:32:29.628 32672-32672/org.covidwatch.tags.android I/Choreographer: Skipped 5378 frames!  The application may be doing too much work on its main thread.
> ```

@BurningAXE had the same issue. Seems like it varies from device to device, because on my device (Samsung A3 2017) it takes 2-3 seconds, and totally not a minute.

I think it is coming from tcn library. @jamesjmtaylor can you pls checkout to this branch, and comment `tcnManager.start()`, `tcnManager.stop()` lines in `InfoBannerViewModel.kt` and see if it helps?

madim committed:
Move tcn starting-stopping logic to its own use case

BurningAXE commented:
> > LGTM. On an unrelated note, has anyone noticed that it takes a full minute before the app begins registering touch events? This is occurring on `develop` as well as this branch. I found this in my logcat logs:
> > ```
> > 2020-04-25 08:32:29.628 32672-32672/org.covidwatch.tags.android I/Choreographer: Skipped 5378 frames!  The application may be doing too much work on its main thread.
> > ```
> 
> @BurningAXE had the same issue. Seems like it varies from device to device, because on my device (Samsung A3 2017) it takes 2-3 seconds, and totally not a minute.
> 
> I think it is coming from tcn library. @jamesjmtaylor can you pls checkout to this branch, and comment `tcnManager.start()`, `tcnManager.stop()` lines in `InfoBannerViewModel.kt` and see if it helps?

@madim Didn't find that file but commented the content of `ensureTcnIsStarted()` in `HomeViewModel` and now the app starts for 2-3 seconds instead of a minute! @Apisov maybe we should create an issue in the TCN repo?

madim committed:
Merge branch 'develop' of https://github.com/covid19risk/covidwatch-android into bugfix/tested-positve-banner

# Conflicts:
#	app/src/main/java/org/covidwatch/android/di/AppModule.kt
#	app/src/main/java/org/covidwatch/android/presentation/HomeFragment.kt
#	app/src/main/java/org/covidwatch/android/presentation/home/HomeViewModel.kt
#	app/src/main/res/layout/fragment_home.xml

nkumarcc reviewed (approved):
- Multi banner worked great
- Stuck waiting for my phone to show it connected with the other that it tested positive, even after Pull to Refresh. Not sure if that's a bug in this PR or a different one, but multibanner looks good and it shows the correct "I've reported UI". - **This issue is in `develop` as well**

Also I didn't comment out that code and it started pretty quickly for me.
